source/output_formatter.py

```diff
@@ -0,0 +1,39 @@
+from os import fsync
+from typing import List
+
+from interface import OutputFormatterInterface, OutputData
+
+
+class OutputFormatterFasta(OutputFormatterInterface):
+
+    def write_output(self, output_data: List[OutputData]) -> None:
+
+        for output in output_data:
+
+            (read_id, _, _, _, _, basecalled_seq, _) = output
+
+            if len(basecalled_seq) == 0:
+                return
+
+            print(">%s" % read_id, file=self.output_stream)
```

Comment:
what-the? How could an "interface" have self.output_stream? :thinking:

Reply:
In python there's not really such strict concept interface. Here I preferred code reuse in baseclass over the interface nature. So OutputFormatterInterface defines public API and also implements som common __init()__ method for all the subclasses. It could be discussed though, what are your ideas?

Reply:
> not really such strict concept interface.
If you *name* something as interface, it better be interface

Reply:
Fair enough.